app/ante.go

```diff
@@ -29,6 +32,7 @@ func NewAnteHandler(
 	mempoolFeeDecorator := txfeeskeeper.NewMempoolFeeDecorator(*txFeesKeeper, mempoolFeeOptions)
 	return sdk.ChainAnteDecorators(
 		ante.NewSetUpContextDecorator(), // outermost AnteDecorator. SetUpContext must be called first
+		wasmkeeper.NewLimitSimulationGasDecorator(wasmConfig.SimulationGasLimit),
```

Comment:
@ethanfrey the [`CountTXDecorator`](https://github.com/CosmWasm/wasmd/blob/master/app/ante.go#L49) is missing in the list. The position data is passed to the contracts via [env](https://github.com/CosmWasm/wasmvm/blob/main/types/env.go#L13) and should be provided to be fully compatible.